### PR TITLE
Reset notebooks to a clean state

### PR DESCRIPTION
**Pull request by kne42**
Reset notebooks under the `examples` folder to a clean ("unrun") state.

### CODE REVIEW

File: examples/layers.ipynb [hunk 1]
```diff
@@ -251,9 +187,9 @@
  ],
  "metadata": {
   "kernelspec": {
-   "display_name": "Python 3",
+   "display_name": "Python [conda env:napari-gui]",
```
Comment (kne42) [inline examples/layers.ipynb:190]: does anyone know if I can just delete this or should I reset it to "reasonable" defaults?

**Review by sofroniewn — APPROVED** [review]
Looks good to me - did you do this by selection `all outputs` `clear` from the cells drop down menu in the notebook? I guess it makes sense to do this before ever recommitting the notebooks

**Comment by kne42** [comment]
@sofroniewn Usually I `reset kernel and clear all outputs` but that works too! If it's inconvenient, you can always bind it to a shortcut (for lab use [jupyterlab/jupyterlab-shortcutui](https://github.com/jupyterlab/jupyterlab-shortcutui)).

**Comment by AhmetCanSolak** [comment]
I also observed that notebook files are changing its content more than addressed in this PR, like execution counts and output texts and so on. Still i believe it is a bad idea to have notebooks inside the repository. We can have notebooks for examples but better to serve somewhere else.

**Comment by sofroniewn** [comment]
There are a couple tools out there to make working with jupyter notebooks in github repos easier - one was made by a software engineer at czi, https://github.com/ttung/nbencdec, and is used in a couple of our other projects - though I know there are a couple other, potentially more popular, solutions out there - I just can't find them right now. Might be worth trying to integrate one of them

**Comment by jni** [comment]
I believe this is the "official" mechanism for dealing with notebooks in git:
https://github.com/jupyter/nbdime

**Comment by AhmetCanSolak** [comment]
@jni @sofroniewn either of the packages will be helpful enough but just will add one more dependency. I personally prefer `nbdime` as it is published under jupyter workspace.

**Comment by sofroniewn** [comment]
Yep - I'm all for using the `jupyter` version if it has the needed functionality. I'm not quite sure why the person from czi wrote the other one in that case, but can ask him sometime too. Maybe it was something about configuring automatic hooks with github?

**Comment by jni** [comment]
Well, there's still advantages to nbencdec (which should be renamed to nbcodec imho =P), namely that things like the github UI still don't render nb diffs nicely, so you might want to catch the messy json early in the process rather than committing it.

Another option is notedown:
https://github.com/aaren/notedown

We wrote Elegant SciPy with it and liked it. However our use case was markdown -> notebook, not the reverse, so I don't know at all how well developed the reverse link is.

**Comment by AhmetCanSolak** [comment]
@jni can you briefly try to reverse way with out notebooks in examples folder and let us know if it is working well. I prefer to go with it as one of us already familiar with tool, will make things quicker.

**Comment by sofroniewn** [comment]
asked the `nbencdec ` dev and he said the big difference is `nbdime` doesn’t support going from `.py` to `.ipynb`, which they use when they refactor code

**Comment by sofroniewn** [comment]
Here's the one I was trying to remember https://github.com/mwouts/jupytext seems quite popular >1000 stars

**Comment by jni** [comment]
@sofroniewn :heart_eyes: that GIF demo is something else! I'm sold.

**Comment by kne42** [comment]
Closing in favor of #67

**Pull request closed without merging**